src/main/scala/advent/solutions/Day3.scala

```diff
@@ -81,17 +240,120 @@ object Day3 {
       *         or [[None]] if they don't intersect at all
       */
     def numberOfStepsToIntersection(wire0: Wire, wire1: Wire): Option[Int] = {
-      ???
+
+      val out = intersectionPoints(wire0, wire1) match {
```

Comment:
`allCoordinates` is being called in `intersectionPoints` and in `countNumberOfStepsToAnIntersection`.  This is a bit error prone, as we might mistakenly calculate them differently in each case.

Why don't you move the calculation to here instead?

```scala
val wire0Coordinates = allCoordinates(wire0)
val wire1Coordinates = allCoordinates(wire1)
intersectionPoints(wire0, wire1).map { points =>
  points.map { point =>
  countNumberOfStepsToAnIntersection(wire0Coordinates, point) + countNumberOfStepsToAnIntersection(wire1Coordinates, point) 
}.min
}
```

I've also made use of `map` on `Option` here instead of pattern matching on `Some` and `None`